### PR TITLE
Update SysId Encoder docs

### LINKED ISSUE #1469: Update sysid documentation for new encoder setup
https://github.com/wpilibsuite/sysid/pull/189

### PR DESCRIPTION
**Pull request by Piphi5**
Fixes #1469

### CODE REVIEW

File: source/docs/software/pathplanning/robot-characterization/configuring-project.rst [hunk 1]
```diff
@@ -31,7 +33,7 @@ Each motor port contains the following parts:
...
+The :guilabel:`Encoder Selection` allows you to configure the encoders that will be used. The types of encoder you can use will vary with the first motor controller type specified (see note :ref:`above <docs/software/pathplanning/robot-characterization/configuring-project:Motor Controller Selection>`), however there are two types that are consistent across all motor controller selections: ``roboRIO`` which corresponds to any encoders that are plugged into the rio, ``CANCoder`` which corresponds to the CTRE CANCoder. If you are using a TalonSRX or TalonFX for your motor controller, the ``Built-In`` corresponds to any quadrature encoder plugged into the encoder port and the ``Tachometer`` option correspondsor to the CTRE Tachometer plugged into the port. If you're using a SparkMAX, the ``Encoder Port`` refers to an encoder plugged into the :ref:`encoder port <https://docs.revrobotics.com/sparkmax/feature-description/encoder-port>` and the ``Data Port`` refers to an encoder plugged into the :ref:`data port <https://docs.revrobotics.com/sparkmax/feature-description/data-port>`. For the Venom, `Built-in` refers to an encoder plugged into the Venom's encoder port.
```
Comment (calcmogul) [inline source/docs/software/pathplanning/robot-characterization/configuring-project.rst:36]:
```suggestion
The :guilabel:`Encoder Selection` allows you to configure the encoders that will be used. The types of encoder you can use will vary with the first motor controller type specified (see note :ref:`above <docs/software/pathplanning/robot-characterization/configuring-project:Motor Controller Selection>`). However, there are two types that are consistent across all motor controller selections: ``roboRIO`` which corresponds to any encoders that are plugged into the rio, ``CANCoder`` which corresponds to the CTRE CANCoder. If you are using a TalonSRX or TalonFX for your motor controller, the ``Built-In`` corresponds to any quadrature encoder plugged into the encoder port and the ``Tachometer`` option corresponds to the CTRE Tachometer plugged into the port. If you're using a Spark MAX, the ``Encoder Port`` refers to an encoder plugged into the :ref:`encoder port <https://docs.revrobotics.com/sparkmax/feature-description/encoder-port>` and the ``Data Port`` refers to an encoder plugged into the :ref:`data port <https://docs.revrobotics.com/sparkmax/feature-description/data-port>`. For the Venom, `Built-in` refers to an encoder plugged into the Venom's encoder port.
```

**Review by calcmogul — APPROVED** [review]

File: source/docs/software/pathplanning/robot-characterization/configuring-project.rst [hunk 2]
```diff
@@ -31,7 +33,7 @@ Each motor port contains the following parts:
...
+The :guilabel:`Encoder Selection` allows you to configure the encoders that will be used. The types of encoder you can use will vary with the first motor controller type specified (see note :ref:`above <docs/software/pathplanning/robot-characterization/configuring-project:Motor Controller Selection>`). However, there are two types that are consistent across all motor controller selections: ``roboRIO`` which corresponds to any encoders that are plugged into the rio, ``CANCoder`` which corresponds to the CTRE CANCoder. If you are using a TalonSRX or TalonFX for your motor controller, the ``Built-in`` corresponds to any quadrature encoder plugged into the encoder port and the ``Tachometer`` option corresponds to the CTRE Tachometer plugged into the port. If you're using a Spark MAX, the ``Encoder Port`` refers to an encoder plugged into the `encoder port <https://docs.revrobotics.com/sparkmax/feature-description/encoder-port>`__ and the ``Data Port`` refers to an encoder plugged into the `data port <https://docs.revrobotics.com/sparkmax/feature-description/data-port>`__. For the Venom, ``Built-in`` refers to an encoder plugged into the Venom's encoder port.
```
Comment (Daltz333) [inline source/docs/software/pathplanning/robot-characterization/configuring-project.rst:36]: I think the latter part of this paragraph can get broken up into a list

File: source/docs/software/pathplanning/robot-characterization/configuring-project.rst [hunk 3]
```diff
@@ -31,11 +33,20 @@ Each motor port contains the following parts:
...
 ^^^^^^^^^^^^^^^^^
-The :guilabel:`Encoder Selection` allows you to configure the encoders that will be used. You can choose the type of encoder with :guilabel:`Encoder`. ``roboRIO`` corresponds to any encoders that are plugged into the rio, ``CANCoder / Alternate`` corresponds to the CTRE CANCoder or any encoders plugged into the Spark Max's Alternate Port, and ``Built-In`` corresponds to any encoder plugged into a CTRE Smart Motor Controller or the encoder port on a Spark Max.
+The :guilabel:`Encoder Selection` allows you to configure the encoders that will be used. The types of encoder you can use will vary with the first motor controller type specified (see note :ref:`above <docs/software/pathplanning/robot-characterization/configuring-project:Motor Controller Selection>`).
 
 .. image:: images/encoder-selection.png
    :alt: Encoder selection section.
 
+Encoder Types
+"""""""""""""
+- **General Types** (consistent across all motor controller selections): ``roboRIO`` corresponds to any encoders that are plugged into the rio, ``CANCoder`` corresponds to the CTRE CANCoder.
```
Comment (calcmogul) [inline source/docs/software/pathplanning/robot-characterization/configuring-project.rst:43]:
```suggestion
- **General Types** (consistent across all motor controller selections): ``roboRIO`` corresponds to any encoders that are plugged into the roboRIO, ``CANCoder`` corresponds to the CTRE CANCoder.
```

**Review by calcmogul — CHANGES_REQUESTED** [review]

File: source/docs/software/pathplanning/robot-characterization/configuring-project.rst [hunk 4]
```diff
@@ -31,11 +33,20 @@ Each motor port contains the following parts:
...
+The :guilabel:`Encoder Selection` allows you to configure the encoders that will be used. The types of encoder you can use will vary with the first motor controller type specified (see note :ref:`above <docs/software/pathplanning/robot-characterization/configuring-project:Motor Controller Selection>`).
 
 .. image:: images/encoder-selection.png
    :alt: Encoder selection section.
 
+Encoder Types
+"""""""""""""
+- **General Types** (consistent across all motor controller selections): ``roboRIO`` corresponds to any encoders that are plugged into the roboRIO, ``CANCoder`` corresponds to the CTRE CANCoder.
+- **TalonSRX** or **TalonFX**: ``Built-in`` corresponds to any quadrature encoder plugged into the encoder port, ``Tachometer`` corresponds to the CTRE Tachometer plugged into the encoder port.
```
Comment (calcmogul) [inline source/docs/software/pathplanning/robot-characterization/configuring-project.rst:44]: The built-in encoder for the Talon FX isn't actually a quadrature encoder.
```cpp
  if (encoderType == "Built-in") {
    if (starts_with(controllerName, "Talon")) {
      FeedbackDevice feedbackDevice;
      if (controllerName == "TalonSRX") {
        feedbackDevice = FeedbackDevice::QuadEncoder;
      } else {
        feedbackDevice = FeedbackDevice::IntegratedSensor;
      }
```

**Review by calcmogul — CHANGES_REQUESTED** [review]

**Review by calcmogul — DISMISSED** [review]

**Review by prateekma — APPROVED** [review]

**Review by Daltz333 — APPROVED** [review]

File: source/docs/software/pathplanning/robot-characterization/configuring-project.rst [hunk 5]
```diff
@@ -31,11 +33,21 @@ Each motor port contains the following parts:
...
 Encoder Selection
 ^^^^^^^^^^^^^^^^^
-The :guilabel:`Encoder Selection` allows you to configure the encoders that will be used. You can choose the type of encoder with :guilabel:`Encoder`. ``roboRIO`` corresponds to any encoders that are plugged into the rio, ``CANCoder / Alternate`` corresponds to the CTRE CANCoder or any encoders plugged into the Spark Max's Alternate Port, and ``Built-In`` corresponds to any encoder plugged into a CTRE Smart Motor Controller or the encoder port on a Spark Max.
+The :guilabel:`Encoder Selection` allows you to configure the encoders that will be used. The types of encoder you can use will vary with the first motor controller type specified (see note :ref:`above <docs/software/pathplanning/robot-characterization/configuring-project:Motor Controller Selection>`).
 
 .. image:: images/encoder-selection.png
    :alt: Encoder selection section.
 
+Encoder Types
+"""""""""""""
```
Comment (Daltz333) [inline source/docs/software/pathplanning/robot-characterization/configuring-project.rst:42]:
```suggestion
~~~~~~~~~~~~~
```

**Review by Daltz333 — CHANGES_REQUESTED** [review]

File: source/docs/software/pathplanning/robot-characterization/configuring-project.rst (hunk 5 above)
Comment (Daltz333) [reply source/docs/software/pathplanning/robot-characterization/configuring-project.rst]: We should not have more than 4 layers of depth, it is bad flow.

**Comment by Piphi5** [comment]
Are there any changes that need to be made before this is merged?

**Review by Daltz333 — APPROVED** [review]

**Pull request merged**